lib/Controller/NotesController.php

```diff
@@ -296,4 +303,47 @@ public function destroy(int $id) : JSONResponse {
 			return [];
 		});
 	}
+
+	/**
+	 * With help from: https://github.com/nextcloud/cookbook
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 * @return DataResponse|FileDisplayResponse
+	 */
+	public function getAttachment(int $noteid, string $path) {
+		try {
+			$note = $this->notesService->get($this->helper->getUID(), $noteid);
+			$notesFolderPath = $this->notesService->getNotesFolder($this->helper->getUID())->getPath();
+			$notePath = $notesFolderPath . "/";
+			if ($note->getCategory() != "") {
+				$notePath .= $note->getCategory() . "/";
+			}
+
+			// calculate how many parentnodes we need to get 'up'
+			$parentcount = substr_count($path, '../');
+			$path = str_replace("../", "", $path);
+
+			$relativeImageNode = $this->root->get($notePath);
+			for ($i = 0; $i < $parentcount; $i++) {
+				$relativeImageNode = $relativeImageNode->getParent();
+			}
+			$targetimage = $this->root->get($relativeImageNode->getPath()."/".$path);
```

Comment:
There must be a check, if we are still in the notes path. Currently, I can load images from other users!

Reply:
ohh yeah, just to understand how you accessed this:

did you prepend '../' the exact same amount to logically get back to "root"? and then append a full, internal nextcloud-path including the different user?

Reply:
Yes

Reply:
Could you provide me with your testsetup? 

I know why and how to fix it, but i cannot replicate it since i always get a second / in my path
Eg.: `?path=../../../test/files/Nextcloud Manual.pdf` results in an exception to be thrown, not an image to be served.

Reply:
I think i fixed it, but since i could not properly reproduce it i cant really verify it. Please try to break it again! I want to be sure this cannot be abused

Reply:
My test setup:
User `A` has a note in `notes/test/test.md` with the following content:
```
![Attack](../../../../B/files/DSC_0020.jpg)
```
User `B` has the image file `DSC_0020.jpg` in his user directory.

I will test your changes soon.

Reply:
Thanks, looks like your fix is working. I think your solution on how to fix this is a good approach! :+1: